Storage/ZoomLevelStore.swift

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+import Common
+
+public struct DomainZoomLevel: Codable, Equatable {
+    let host: String
+    public let zoomLevel: CGFloat
+
+    public init(host: String, zoomLevel: CGFloat) {
+        self.host = host
+        self.zoomLevel = zoomLevel
+    }
+}
+
+public class ZoomLevelStore {
+    public static let shared = ZoomLevelStore()
+
+    private(set) var domainZoomLevels = [DomainZoomLevel]()
+    private var logger: Logger
+
+    private static let fileName = "domain-zoom-levels"
+    private static let pathExtension = "json"
+
+    private let url = URL(fileURLWithPath: fileName,
+                          relativeTo: FileManager.documentsDirectoryURL).appendingPathExtension(pathExtension)
+
+    private init(logger: Logger = DefaultLogger.shared) {
+        self.logger = logger
+        domainZoomLevels = loadAll()
+    }
+
+    public func save(_ domainZoomLevel: DomainZoomLevel) {
+        if let dzl = findZoomLevel(forDomain: domainZoomLevel.host),
```

Comment:
nit, I find it easier to read when variables have their full name instead of abbreviations. I think you had a name conflict using the full name, so if I get the code right it can be named like this?  
```suggestion
        if let foundDomainZoomLevel = findZoomLevel(forDomain: domainZoomLevel.host),
```